### PR TITLE
[BEAM-498] Inline DoFnRunnerBase and fix up

### PR DESCRIPTION
**Pull request by kennknowles**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

This class had just one subclass and cleaning up these little bits makes it clearer as we move to the new `DoFn`. Nothing radical, but I need to make `DoFnRunners` work with old and new `DoFn` so each dangling class just adds complexity to the field of operation.

R: @peihe

### CODE REVIEW

**Comment by kennknowles** [comment]
Yes, that is in progress.

**Comment by peihe** [comment]
LGTM

nit:
remove the comment of ListOutputManager in here:
https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java#L774

**Comment by kennknowles** [comment]
R: @amitsela as a randomly selected committer (@peihe touched this code most recently so I got his input first)

File: runners/core-java/src/main/java/org/apache/beam/runners/core/DoFnRunnerBase.java [hunk 1]
```diff
@@ -1,559 +0,0 @@
...
-import org.apache.beam.sdk.values.TupleTag;
-import org.joda.time.Instant;
-import org.joda.time.format.PeriodFormat;
-
-/**
- * A base implementation of {@link DoFnRunner}.
- *
- * <p> Sub-classes should override {@link #invokeProcessElement}.
- */
-public abstract class DoFnRunnerBase<InputT, OutputT> implements DoFnRunner<InputT, OutputT> {
```
Comment (amitsela) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/DoFnRunnerBase.java]: `DoFnRunnerBase` is mentioned in [SimpleDoFnRunnerTest JavaDoc](https://github.com/apache/incubator-beam/blob/master/runners/core-java/src/test/java/org/apache/beam/runners/core/SimpleDoFnRunnerTest.java#L39).
Probably want to change that to `SimpleDoFnRunner`/`DoFnRunner`.

**Comment by amitsela** [comment]
And thanks 😉, things sometimes move fast around here and that's a very efficient way to bring things to my attention.
I'll start playing around with this and see how to use it in the runner - maybe even with as part of #1055

**Pull request merged**